### ISSUE TITLE
webpanel dont show miner list

### ISSUE THREAD

**Issue by hara2214** [UnamSanctam/UnamWebPanel#190]

**Comment by UnamSanctam** [comment]
1. Make sure that your files have the correct permissions for your setup, you can test if it's able to write to the database by going to the "Configurations" tab and then adding a new configuration (you can use `{}` as the data), if you can then find it in the "Edit Configurations" list then it can write, if not then it can't. Also make sure that you're not forcing redirections from `http` to `https`. Also make sure that the miner is actually running by using the checker.
2. It's to change the miners configurations (settings), it's formatted exactly the same as the "Remote Configuration":
https://github.com/UnamSanctam/SilentCryptoMiner/wiki/ETH-Miner#remote-configuration
https://github.com/UnamSanctam/SilentCryptoMiner/wiki/XMR-Miner#remote-configuration

**Comment by hara2214** [comment]
How do I verify that I have permissions? I opened the web server using xampp in ubuntu 18. And I tried using both http and https, but I failed. I added the configuration, but it doesn't seem to be saved in the database. It doesn't come up.

![image](https://user-images.githubusercontent.com/112808776/188343405-2b6a6ba0-3989-45ea-bb9c-f8a9045579e5.png)

![image](https://user-images.githubusercontent.com/112808776/188343421-977370bb-3535-4985-b39b-5249c73f24d0.png)

![image](https://user-images.githubusercontent.com/112808776/188343474-bb5e7851-ef82-44ee-bbca-e1a15b388791.png)

**Comment by UnamSanctam** [comment]
Yes then it's not able to write to your database (inside `db\unamwebpanel.db`), you'll have to set up the correct write permissions for your webserver user, in Linux you (usually) have to set up all the permissions.

**Comment by destylul** [comment]
> Yes then it's not able to write to your database (inside `db\unamwebpanel.db`), you'll have to set up the correct write permissions for your webserver user, in Linux you (usually) have to set up all the permissions.

Uhhh, I have the same problem as him but I can add
![Screenshot_2023-01-09-16-16-42-44](https://user-images.githubusercontent.com/111581758/211275055-fe395dbb-99ce-4998-a47e-055f5f8e6a08.png)
![Screenshot_2023-01-09-16-16-32-36](https://user-images.githubusercontent.com/111581758/211275066-90d466d1-5901-4688-bdd1-138a3c8c2d75.png)

**Comment by UnamSanctam** [comment]
>Uhhh, I have the same problem as him but I can add

Make sure that your webhost isn't forcing JavaScript or something like it, and make sure to only use https if you have a certificate (like your last site).

**Comment by destylul** [comment]
> > Uhhh, I have the same problem as him but I can add
> 
> Make sure that your webhost isn't forcing JavaScript or something like it, and make sure to only use https if you have a certificate (like your last site).

my site doesn't have ssl

**Comment by destylul** [comment]
> > Uhhh, I have the same problem as him but I can add
> 
> Make sure that your webhost isn't forcing JavaScript or something like it, and make sure to only use https if you have a certificate (like your last site).

If I make my site private, will the web panel work or not?

**Comment by UnamSanctam** [comment]
>my site doesn't have ssl

Then make sure you use http when you enter the URL.

>If I make my site private, will the web panel work or not?

You mean some sort of private option on your webhost? Then most likely nothing will be able to connect to it I would assume.

**Comment by destylul** [comment]
> > my site doesn't have ssl
> 
> Then make sure you use http when you enter the URL.
> 
> > If I make my site private, will the web panel work or not?
> 
> You mean some sort of private option on your webhost? Then most likely nothing will be able to connect to it I would assume.


I want my site to be private to keep it from being discovered by the web host creator. (It's useless though.) I think the web panel doesn't show the miners list, which is quite difficult to fix, and I don't know what is the cause.
So what causes my web panel not to show the list of miners?...I shouldn't have asked you about these issues, and I think the best way is to stop using WebPanel.

**Comment by UnamSanctam** [comment]
>I want my site to be private to keep it from being discovered by the web host creator. (It's useless though.) I think the web panel doesn't show the miners list, which is quite difficult to fix, and I don't know what is the cause.
So what causes my web panel not to show the list of miners?...I shouldn't have asked you about these issues, and I think the best way is to stop using WebPanel.

Well you can never hide it from the one hosting it. Make sure that there's no option enabled on your webhost that forces JavaScript to be enabled. And make sure that you're using http:// when entering the URL into the miner (since you said that you have no SSL).

**Comment by destylul** [comment]
> > I want my site to be private to keep it from being discovered by the web host creator. (It's useless though.) I think the web panel doesn't show the miners list, which is quite difficult to fix, and I don't know what is the cause.
> > So what causes my web panel not to show the list of miners?...I shouldn't have asked you about these issues, and I think the best way is to stop using WebPanel.
> 
> Well you can never hide it from the one hosting it. Make sure that there's no option enabled on your webhost that forces JavaScript to be enabled. And make sure that you're using http:// when entering the URL into the miner (since you said that you have no SSL).

Yes, thank you. I will continue to fix the site on my own and will not give up. Have a nice day!

**Comment by UnamSanctam** [comment]
Alright great, hopefully it works out. I hope you have a nice day too.

**Comment by destylul** [comment]
> Alright great, hopefully it works out. I hope you have a nice day too.

I just started editing the website now because I'm too busy in real life. I have enabled web error display and I see these lines, do you know the cause?
![IMG_20230109_221017](https://user-images.githubusercontent.com/111581758/211340760-bee1e120-f70a-47da-9757-c434f86b6155.jpg)

**Comment by UnamSanctam** [comment]
Are you sure no files are missing (like the `assets/php/safety-check.php` file)? If not then it could also be due to the include path being weird.

**Comment by destylul** [comment]
> Are you sure no files are missing (like the `assets/php/safety-check.php` file)? If not then it could also be due to the include path being weird.

Ah, it's because I'm missing the file. But when I go to the login page and enter the information, I get an error.
![Uploading Screenshot_2023-01-09-22-42-00-37.png…]()

**Comment by destylul** [comment]
> Are you sure no files are missing (like the `assets/php/safety-check.php` file)? If not then it could also be due to the include path being weird.

It's because I loaded files one by one, and maybe I missed something. I tried to transfer all files in, and it worked and my webpanel finally shows the list of miners.

**Comment by UnamSanctam** [comment]
Great.
